app/javascript/chat/ReportAbuse/index.jsx

```diff
@@ -0,0 +1,143 @@
+import { h } from 'preact';
+import PropTypes from 'prop-types';
+import { useState } from 'preact/hooks';
+import { reportAbuse } from '../actions/requestActions';
+import { addSnackbarItem } from '../../Snackbar';
+import { Button } from '@crayons';
+
+function ReportAbuse({ resource: data }) {
+  const [category, setCategory] = useState(null);
+
+  const handleChange = (e) => {
+    setCategory(e.target.value);
+  };
+
+  const handleSubmit = async () => {
+    const response = await reportAbuse(
+      data.message,
+      'connect',
+      category,
+      data.user_id,
+    );
+    const { success, message } = response;
+    if (success) {
+      const confirmBlock = window.confirm(
```

Comment:
Refactor (non-blocking): Use the [Modal](https://storybook.forem.com/?path=/story/3-components-modals--default) component instead. You can use a Button Group component for the OK/Cancel and you can await a boolean value.

If this is too much work for now, we can do this in another PR. In fact we should probably have  additional props on the component for action buttons, cc: @ludwiczakpawel

Reply:
Yea. I'd definitely postpone it. We've been working on some design updates for Connect so I'd rather ship stuff looking not perfect right now and revisit the UI a bit later, definitely in a separate PR.